pubnub-entities/src/main/kotlin/com/pubnub/entities/Space.kt

```diff
@@ -11,30 +21,62 @@ fun PubNub.getSpaces(
     sort: Collection<PNSortKey> = listOf(),
     includeCount: Boolean = false,
     includeCustom: Boolean = false
-) = getAllChannelMetadata(
-    limit = limit, page = page, filter = filter, sort = sort, includeCount = includeCount, includeCustom = includeCustom
-)
+): ExtendedRemoteAction<PNSpaceArrayResult?> = firstDo(
+    getAllChannelMetadata(
+        limit = limit,
+        page = page,
+        filter = filter,
+        sort = sort,
+        includeCount = includeCount,
+        includeCustom = includeCustom
+    )
+).then {
+    map(
+        it,
+        PNOperationType.SpaceOperation
+    ) { pnChannelMetadataArrayResult -> pnChannelMetadataArrayResult.toPNSpaceArrayResult() }
+}
 
 fun PubNub.getSpace(
-    space: String,
+    spaceId: String,
     includeCustom: Boolean = false
-) = getChannelMetadata(
-    channel = space,
-    includeCustom = includeCustom
-)
+): ExtendedRemoteAction<PNSpaceResult?> = firstDo(
+    getChannelMetadata(channel = spaceId, includeCustom = includeCustom)
+).then {
+    map(
+        it,
+        PNOperationType.SpaceOperation
+    ) { pnChannelMetadataResult -> pnChannelMetadataResult.toPNSpaceResult() }
+}
 
 fun PubNub.setSpace(
     space: String,
```

Comment:
Should this not be:

```suggestion
    spaceId: String
```

Reply:
Good catch. It has been fixed already and is available on master.